build/windows/contentScope.js

```diff
@@ -0,0 +1,4424 @@
+            return false
+        }
+        const stack = getStack();
+        const errorFiles = getStackTraceUrls(stack);
+        for (const path of errorFiles) {
+            if (shouldExemptUrl(type, path.href)) {
+                return true
+            }
+        }
+        return false
+    }
+
+    function camelcase (dashCaseText) {
+        return dashCaseText.replace(/-(.)/g, (match, letter) => {
+            return letter.toUpperCase()
+        })
+    }
+
+    function getStack () {
+        return new Error().stack
+    }
+
+    /**
+     * @template {object} P
+     * @typedef {object} ProxyObject<P>
+     * @property {(target?: object, thisArg?: P, args?: object) => void} apply
+     */
+
+    /**
+     * @template [P=object]
+     */
+    class DDGProxy {
+        /**
+         * @param {string} featureName
+         * @param {P} objectScope
+         * @param {string} property
+         * @param {ProxyObject<P>} proxyObject
+         */
+        constructor (featureName, objectScope, property, proxyObject) {
+            this.objectScope = objectScope;
+            this.property = property;
+            this.featureName = featureName;
+            this.camelFeatureName = camelcase(this.featureName);
+            const outputHandler = (...args) => {
+                const isExempt = shouldExemptMethod(this.camelFeatureName);
+                // The normal return value
+                if (isExempt) {
+                    return DDGReflect.apply(...args)
+                }
+                return proxyObject.apply(...args)
+            };
+            {
+                this._native = objectScope[property];
+                const handler = {};
+                handler.apply = outputHandler;
+                this.internal = new globalObj.Proxy(objectScope[property], handler);
+            }
+        }
+
+        // Actually apply the proxy to the native property
+        overload () {
+            {
+                this.objectScope[this.property] = this.internal;
+            }
+        }
+    }
+
+    let DDGReflect;
+
+    // Exports for usage where we have to cross the xray boundary: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Sharing_objects_with_page_scripts
+    {
+        DDGReflect = globalObj.Reflect;
+    }
+
+    /* global Bluetooth, Geolocation, HID, Serial, USB */
+
+    function initPermissionsInUseDetection () {
+        const featureName = 'permission-usage';
+
+        const Permission = {
+            Geolocation: 'geolocation',
+            Camera: 'camera',
+            Microphone: 'microphone'
+        };
+
+        const Status = {
+            Inactive: 'inactive',
+            Accessed: 'accessed',
+            Active: 'active',
+            Paused: 'paused'
+        };
+
+        const isFrameInsideFrame = window.self !== window.top && window.parent !== window.top;
+
+        function windowsPostMessage (name, data) {
```

Comment:
I believe this should also be using the secure messaging methods: https://github.com/more-duckduckgo-org/windows-browser/blob/develop/WindowsBrowser/Scripts/SecureContentScriptSandboxTemplate.js

Reply:
I see what you're saying, still a little bit misleading, but I can't think of a better way than having those things at the same location, which seems like a no-go.